processor/filterprocessor/factory.go

```diff
@@ -28,6 +28,8 @@ const (
 	typeStr = "filter"
 )
 
+var processorCapabilities = component.ProcessorCapabilities{MutatesConsumedData: false}
+
 // NewFactory returns a new factory for the Filter processor.
 func NewFactory() component.ProcessorFactory {
 	return processorhelper.NewFactory(
```

Comment:
Instead of:
```go
processorhelper.WithMetrics(createMetricsProcessor)
```

Would it make sense to consider working towards:
```go
processorhelper.WithMetrics(filterMetrics, processorhelper.WithCapabilities(processorCapabilities))
```

And eventually remove the need for `processorhelper/processor.go`.

Not sure if that's feasible given the rest of the startup code & need for backwards compatibilty.

Reply:
I will think about, but seems hard to cover all the cases:
1. Somewhere you need a new instance of the underlying processor that is wrapped so there needs to be a function that does what the current create does before calling into the new helper.
2. Capabilities may differ based on the config or params (things that are passed during create method). I know we don't do that now but it is a possibility.